Student/SubmitAssignment/Assignments/AssignmentsViewController.swift

```diff
@@ -44,27 +51,35 @@ class AssignmentsViewController: UIViewController, AssignmentsView {
         activityIndicator.hidesWhenStopped = true
         addNavigationButton(UIBarButtonItem(customView: activityIndicator), side: .right)
 
-        presenter?.viewIsReady()
+        assignments.refresh(force: true)
     }
 
     func update() {
         tableView.reloadData()
-        presenter?.assignments.pending == true ? activityIndicator.startAnimating() : activityIndicator.stopAnimating()
+        assignments.pending == true ? activityIndicator.startAnimating() : activityIndicator.stopAnimating()
     }
 }
 
 extension AssignmentsViewController: UITableViewDataSource, UITableViewDelegate {
     func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
-        return presenter?.assignments.count ?? 0
+        var count = assignments.count
+        if assignments.hasNextPage == true {
+            count += 1
+        }
+        return count
     }
 
     func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        if indexPath.row == assignments.count {
+            assignments.getNextPage()
+            return LoadingCell(style: .default, reuseIdentifier: nil)
+        }
```

Comment:
nice trick. Should we be using this instead of the scrollViewDidScroll as our common pattern?

Reply:
Yeah, I think so. It handles more cases and it's a nice way to show that we're loading more.